akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala

```diff
@@ -85,6 +87,16 @@ class FlowSplitWhenSpec extends AkkaSpec {
       }
     }
 
+    "not emit substreams if the parent stream is empty" in assertAllStagesStopped {
+
+      Await.result(
+        Source.empty[Int]
+          .splitWhen(_ ⇒ true).lift
+          .mapAsync(1)(_.runWith(Sink.headOption)).grouped(10).runWith(Sink.headOption),
+        3.seconds) should ===(None)
```

Comment:
If an empty substream is emitted this reqults in a `Some(None)`. A bit tricky test, but nice and compact.